amd_openvx_extensions/amd_nn/src/batch_normalization_layer.cpp

```diff
@@ -172,6 +192,40 @@ static vx_status VX_CALLBACK initializeBatchNormalizationLayer(vx_node node, con
         if (err) return VX_FAILURE;
     }
     ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[6], VX_TENSOR_BUFFER_OPENCL, &data->output_mem, sizeof(data->output_mem)));
+#elif ENABLE_HIP
+    // input and output memory.
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_BUFFER_HIP, &data->input_mem, sizeof(data->input_mem)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[1], VX_TENSOR_BUFFER_HIP, &data->bnMean, sizeof(data->bnMean)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[2], VX_TENSOR_BUFFER_HIP, &data->bnVariance, sizeof(data->bnVariance)));
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[3], VX_TENSOR_BUFFER_HIP, &data->bnScale, sizeof(data->bnScale)));
+    if(parameters[4]) {
+        ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[4], VX_TENSOR_BUFFER_HIP, &data->bnBias, sizeof(data->bnBias)));
+    } else {
+        vx_context vxContext = vxGetContext((vx_reference)node);
+        int hip_device = -1;
+        ERROR_CHECK_STATUS(vxQueryContext(vxContext, VX_CONTEXT_ATTRIBUTE_AMD_HIP_DEVICE, &hip_device, sizeof(hip_device)));
+        if (hip_device < 0) {
+            return VX_FAILURE;
+        }
+        hipError_t errcode_ret = hipSuccess;
+        errcode_ret = hipSetDevice(hip_device);
```

Comment:
Do we need to do this again here since hip_device is already set for the context

Reply:
No, we don't need it. I removed it.